app/app.go

```diff
@@ -367,31 +373,88 @@ func New(
-		module.NewVersionedModule(crisis.NewAppModule(&app.CrisisKeeper, skipGenesisInvariants), v1.Version, v2.Version),
-		module.NewVersionedModule(gov.NewAppModule(appCodec, app.GovKeeper, app.AccountKeeper, app.BankKeeper), v1.Version, v2.Version),
-		module.NewVersionedModule(mint.NewAppModule(appCodec, app.MintKeeper, app.AccountKeeper), v1.Version, v2.Version),
-		module.NewVersionedModule(slashing.NewAppModule(appCodec, app.SlashingKeeper, app.AccountKeeper, app.BankKeeper, app.StakingKeeper), v1.Version, v2.Version),
-		module.NewVersionedModule(distr.NewAppModule(appCodec, app.DistrKeeper, app.AccountKeeper, app.BankKeeper, app.StakingKeeper), v1.Version, v2.Version),
-		module.NewVersionedModule(staking.NewAppModule(appCodec, app.StakingKeeper, app.AccountKeeper, app.BankKeeper), v1.Version, v2.Version),
-		module.NewVersionedModule(evidence.NewAppModule(app.EvidenceKeeper), v1.Version, v2.Version),
-		module.NewVersionedModule(authzmodule.NewAppModule(appCodec, app.AuthzKeeper, app.AccountKeeper, app.BankKeeper, app.interfaceRegistry), v1.Version, v2.Version),
-		module.NewVersionedModule(ibc.NewAppModule(app.IBCKeeper), v1.Version, v2.Version),
-		module.NewVersionedModule(params.NewAppModule(app.ParamsKeeper), v1.Version, v2.Version),
-		module.NewVersionedModule(transfer.NewAppModule(app.TransferKeeper), v1.Version, v2.Version),
-		module.NewVersionedModule(blob.NewAppModule(appCodec, app.BlobKeeper), v1.Version, v2.Version),
-		module.NewVersionedModule(blobstream.NewAppModule(appCodec, app.BlobstreamKeeper), v1.Version, v2.Version),
-		module.NewVersionedModule(upgrade.NewAppModule(app.UpgradeKeeper), v2.Version, v2.Version),
-	)
+	app.mm, err = module.NewManager([]module.VersionedModule{
+		{
+			Module:      genutil.NewAppModule(app.AccountKeeper, app.StakingKeeper, app.BaseApp.DeliverTx, encodingConfig.TxConfig),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      auth.NewAppModule(appCodec, app.AccountKeeper, nil),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      vesting.NewAppModule(app.AccountKeeper, app.BankKeeper),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      bank.NewAppModule(appCodec, app.BankKeeper, app.AccountKeeper),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      capability.NewAppModule(appCodec, *app.CapabilityKeeper),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      feegrantmodule.NewAppModule(appCodec, app.AccountKeeper, app.BankKeeper, app.FeeGrantKeeper, app.interfaceRegistry),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      crisis.NewAppModule(&app.CrisisKeeper, skipGenesisInvariants),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      gov.NewAppModule(appCodec, app.GovKeeper, app.AccountKeeper, app.BankKeeper),
+			FromVersion: v1, ToVersion: v2,
```

Comment:
I like the formatting change a lot

unrelated to this PR, does switching the `ToVersion` here to 0 allow for any future version to work? if so, can we do that fix the testground issues (app version == to some large number to have large block sizes)?

Reply:
> does switching the ToVersion here to 0 allow for any future version to work?

No everything currently must be explicit (and continuous). I think it's better if we have a different mechanism for testing the large block sizes - we could disconnect them from the appversion by simply having a mapping that is introduced in the app constructor